src/ShopifyApp/Services/ChargeHelper.php

```diff
@@ -267,7 +267,7 @@ public function details(Plan $plan, IShopModel $shop): PlanDetailsTransfer
         $transfer->returnUrl = URL::secure(
             Util::getShopifyConfig('billing_redirect'),
             ['plan' => $plan->getId()->toNative()]
-        );
+        )."?shop={$shop->getDomain()->toNative()}";
```

Comment:
@lucasmichot Is there a cleaner way to add a query param to URL::secure, do you know?

Reply:
```suggestion
        ).'?'.http_build_query(['shop' => $shop->getDomain()->toNative()]);
```

Reply:
I would probably use `http_build_query`